pontoon/projects/views.py

```diff
@@ -62,6 +71,90 @@ def ajax_info(request, slug):
     })
 
 
+@permission_required_or_403('base.can_manage_project')
+@transaction.atomic
+@require_AJAX
+def ajax_notifications(request, slug):
+    """Notifications tab."""
+    project = get_object_or_404(Project.objects.available(), slug=slug)
+    available_locales = project.locales.order_by('name')
+
+    # Send notifications
+    if request.method == 'POST':
+        message = request.POST.get('message', None)
+        selected_ids = request.POST.get('selected_locales')
+
+        # TODO: implement validation on client side
+        if not selected_ids:
+            return HttpResponseBadRequest(
+                'no_locales'
+            )
+
+        if not message:
+            return HttpResponseBadRequest(
+                'no_message'
+            )
+
+        contributors = User.objects.filter(
+            translation__entity__resource__project=project,
+        )
+
+        # For performance reasons, only filter contributors for selected
+        # locales if different from all project locales
+        available_ids = sorted(list(available_locales.values_list('id', flat=True)))
+        selected_ids = sorted(map(int, selected_ids.split(',')))
+
+        if available_ids != selected_ids:
+            contributors = User.objects.filter(
+                translation__entity__resource__project=project,
+                translation__locale__in=available_locales.filter(id__in=selected_ids)
+            )
+
+        identifier = uuid.uuid4().hex
+        for contributor in contributors.distinct():
+            notify.send(
+                request.user,
```

Comment:
I wonder if we could keep this here, because every send will cause an insert to database:
https://github.com/django-notifications/django-notifications/blob/master/notifications/models.py#L270
If I'm reading that code correctly.
If there's a lot of contributors, it may be reasonable to defer this task to celery and build UI on top of that.

However, I don't have any benchmarks, maybe that's not the problem.